test/Kestrel.Core.Tests/HttpProtocolFeatureCollectionTests.cs

```diff
@@ -143,24 +151,22 @@ public void FeaturesSetByGenericSameAsByType()
         [Fact]
         public void Http2StreamFeatureCollectionDoesNotIncludeMinRateFeatures()
         {
-            var http2Stream = new Http2Stream(new Http2StreamContext
-            {
-                ServiceContext = new TestServiceContext(),
-                ConnectionFeatures = new FeatureCollection(),
-                TimeoutControl = Mock.Of<ITimeoutControl>(),
-                Transport = Mock.Of<IDuplexPipe>(),
-                ServerPeerSettings = new Http2PeerSettings(),
-                ClientPeerSettings = new Http2PeerSettings(),
-            });
-            var http2StreamCollection = (IFeatureCollection)http2Stream;
-
-            Assert.Null(http2StreamCollection.Get<IHttpMinRequestBodyDataRateFeature>());
-            Assert.Null(http2StreamCollection.Get<IHttpMinResponseDataRateFeature>());
+            Assert.Null(_http2Collection.Get<IHttpMinRequestBodyDataRateFeature>());
+            Assert.Null(_http2Collection.Get<IHttpMinResponseDataRateFeature>());
 
             Assert.NotNull(_collection.Get<IHttpMinRequestBodyDataRateFeature>());
             Assert.NotNull(_collection.Get<IHttpMinResponseDataRateFeature>());
         }
 
+        [Fact]
+        public void Http2StreamFeatureCollectionDoesIncludeUpgradeFeature()
```

Comment:
This is moved to HttpProtocol so is there an equivalent test to ensure it exist in Http1Connection?

Reply:
For Http1Connection, we have tests that actually use the feature.